ReSharper.FSharp/src/FSharp.Psi/src/Util/XmlDocTemplateUtil.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JetBrains.Annotations;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Impl;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Tree;
+using JetBrains.ReSharper.Psi.ExtensionsAPI;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Util
+{
+  public static class XmlDocTemplateUtil
+  {
+    [NotNull]
+    public static string GetDocTemplate(IDocCommentBlock docCommentBlock, Func<int, string> linePrefix, string lineSeparator)
+    {
+      var text = new StringBuilder();
+      var line = 1;
+
+      text.Append($"summary>{lineSeparator}");
+      text.Append($"{linePrefix(line++)}{lineSeparator}");
+      text.Append($"{linePrefix(line++)}</summary>{lineSeparator}");
+
+      foreach (var parameter in GetParameters(docCommentBlock.Parent))
+        text.Append($"{linePrefix(line++)}<param name=\"{parameter}\"></param>{lineSeparator}");
+
+      return text.ToString();
+    }
+
+    private static IEnumerable<string> GetParameters(ITreeNode declaration) =>
+      declaration switch
+      {
+        IBinding binding => binding.Expression is ILambdaExpr lambda
```

Comment:
@DedSec256 This code is very useful and we need it in some other places. Let's move it to some util class (maybe `FSharpParameterUtil` or something, I'm OK with using a better name later) and take other use cases into account?

What we need is getting list of groups of top level parameter names:
```
let f a (b, c)  _ = () // [a]; [b; c]; [???]
let f (e, (f, g) = () // [e; ???]
let f (h, i as j) = () // [j]
```

We could also make the util return list of groups of parameter patterns and get their names separately.

Maybe inheriting the visitor type would also allow removing this long list of type checks.

Reply:
> Let's move it to some util class (maybe FSharpParameterUtil or something) and take some other use cases into account?
> We could also make the util return patterns instead of strings and get their names separately.

yes, it's all in the plans (: